### PR TITLE
Documentation about documentation

### LINKED ISSUE #2485: Tutorials/Documentation contribution
Hi Team,

Love the work you are putting in so far (noticed if /else will be coming in release 0.11... super!)

I would love to contribute tutorials/docs. I have built up some tutorials, using data from the ```rdatatable``` wiki. What is the process for submitting tutorials/ additions to the documentations? 

If you don't mind me asking, when will ```DT[i,j,by()]``` be fully implemented, where one can filter the result of a groupby, and run a computation in the ```j``` column as well?

Cheers

### PR DESCRIPTION
**Pull request by st-pasha**
Added doc page about how to contribute documentation.

Closes #2485

### CODE REVIEW

File: docs/develop/documentation.rst [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+      -[new] New feature that was added
+
+      -[enh] Improvement of an existing feature or function
+
+      -[fix] Bug fix
+
+      -[api] API change
+
+  In addition, if any such item ends with the text of the form ``[#333]``,
+  then this will be automatically converted into a link to a github issue/PR
```
Comment (oleksiyskononenko) [inline docs/develop/documentation.rst:267]: May be surround `/` with spaces just like everywhere else.

File: docs/develop/documentation.rst [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+``static const char* doc_py_rbind``. The content of the latter variable will
+be pre-processed and then rendered as RST. The ``:doc:`` parameter is optional,
+if omitted the directive will attempt to find the docstring automatically.
+
+The optional ``:tests:`` parameter should point to a file where the tests for
+this function are located. This will be included as a link in the rendered
+output.
+
+
+In order to document a getter/setter property of a class, use the following::
```
Comment (oleksiyskononenko) [inline docs/develop/documentation.rst:305]: `getter / setter`

File: docs/develop/documentation.rst [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+        :settable: newkey
+        :deletable:
+
+The ``:src:`` parameter can now accept two function names: the getter and the
+setter. In addition, the ``:settable:`` parameter will have the name of the setter
+value as it will be displayed in the docs. Lastly, ``:deletable:`` marks this
+class property as deletable.
+
+
+The docstring of the function/method/etc is preprocessed before it is rendered
```
Comment (oleksiyskononenko) [inline docs/develop/documentation.rst:320]: `function / method / etc`

File: docs/develop/documentation.rst (hunk 1 above)
Comment (st-pasha) [reply docs/develop/documentation.rst:267]: According to [Chicago Manual of Style](https://www.chicagomanualofstyle.org/qanda/data/faq/topics/Punctuation/faq0096.html) there should be no spaces when `/` separates individual words (as opposed to sentences).

Comment (oleksiyskononenko) [reply docs/develop/documentation.rst:267]: I see, then let's make it be consistent with this manual in the other places like this one https://github.com/h2oai/datatable/pull/2492/files#diff-dfe271025faea4bfbd4bddd37499bcd7R285 ?

**Pull request merged**